platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/GossipEvent.java

```diff
@@ -23,22 +23,39 @@
 import com.swirlds.common.system.events.BaseEventUnhashedData;
 import com.swirlds.platform.EventStrings;
 import com.swirlds.platform.gossip.chatter.protocol.messages.ChatterEvent;
+import edu.umd.cs.findbugs.annotations.NonNull;
 import java.io.IOException;
 import java.time.Instant;
 import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.IntUnaryOperator;
 
 /**
  * A class used to hold information about an event transferred through gossip
  */
 public class GossipEvent implements EventIntakeTask, BaseEvent, ChatterEvent {
     private static final long CLASS_ID = 0xfe16b46795bfb8dcL;
+
+    /**
+     * Lambda to update the intake counter when an event exits the intake pipeline. The lambda decrements the counter,
+     * but prevents it from going below 0.
+     */
+    private static final IntUnaryOperator EXIT_INTAKE = count -> count > 0 ? count - 1 : 0;
+
     private static final long ROUND_CREATED_UNDEFINED = -1;
     private BaseEventHashedData hashedData;
     private BaseEventUnhashedData unhashedData;
     private EventDescriptor descriptor;
     private Instant timeReceived;
     private long roundCreated = ROUND_CREATED_UNDEFINED;
 
+    /**
+     * A reference to the atomic integer tracking how many events the sender currently has in the intake pipeline.
+     * <p>
+     * If an event wasn't received through gossip, then this will be null.
+     */
+    private AtomicInteger intakeCounter;
```

Comment:
Why does an event need to have a reference to this?

Reply:
I think its a lot better if we simply keep a reference who we received it from

Reply:
If you only keep a reference to who we received it from, then every location where an event may exit the intake pipeline must have a reference to the event intake counter object. Having the atomic int directly means all you need is the gossip event to exit the pipeline

Given how convoluted the intake pipeline is, the change to directly store the int represented a major simplification

Reply:
I understand, but this, in my mind, is introducing more technical debt

Reply:
Atomic int has been removed from the gossip event, in favor of just tracking event sender.